esrally/mechanic/telemetry.py

```diff
@@ -410,14 +410,17 @@ def record_stats_per_index(self, name, stats):
         :param stats: A dict with returned CCR stats for the index.
         """
 
-        for shard_num, shard_stats in stats.items():
-            shard_metadata = {
-                "cluster": self.cluster_name,
-                "index": name,
-                "shard": shard_num
-            }
+        for shard_stats in stats:
+            shard_id = shard_stats.get("shard_id", None)
+            if shard_id is not None:
```

Comment:
Probably a matter of taste but you could also write this as `if "shard_id" in shard_stats` and then use `shard_stats["shard_id"]` in the body.

Reply:
Thanks for the suggestion I prefer this as well; addressed in 581a69f85ed67f9f0d363f3166d3683c7de36ca0